ui/page/dcrdex/market.go

```diff
@@ -0,0 +1,937 @@
+package dcrdex
+
+import (
+	"fmt"
+	"image/color"
+	"strconv"
+
+	"gioui.org/font"
+	"gioui.org/layout"
+	"gioui.org/unit"
+	"gioui.org/widget"
+
+	"github.com/crypto-power/cryptopower/app"
+	libutils "github.com/crypto-power/cryptopower/libwallet/utils"
+	"github.com/crypto-power/cryptopower/ui/cryptomaterial"
+	"github.com/crypto-power/cryptopower/ui/load"
+	"github.com/crypto-power/cryptopower/ui/modal"
+	"github.com/crypto-power/cryptopower/ui/page/components"
+	"github.com/crypto-power/cryptopower/ui/utils"
+	"github.com/crypto-power/cryptopower/ui/values"
+)
+
+const (
+	DEXMarketPageID = "dex_market"
+	// maxOrderDisplayedInOrderBook is the maximum number of orders that can be
+	// accommodated/displayed on the order book.
+	maxOrderDisplayedInOrderBook = 7
+)
+
+var (
+	dp5                        = values.MarginPadding5
+	dp8                        = values.MarginPadding8
+	dp300                      = values.DP300
+	orderFormAndOrderBookWidth = (values.AppWidth / 2) - 40 // Minus 40 px to allow for margin between the order form and order book.
+	// orderFormAndOrderBookHeight is a an arbitrary height that accommodates
+	// the current order form elements and maxOrderDisplayedInOrderBook. Use
+	// this to ensure they(order form nad orderbook) have the same height as
+	// they are displayed sided by side.
+	orderFormAndOrderBookHeight = values.MarginPadding515
+
+	limitOrderIndex = 0
+	orderTypes      = []cryptomaterial.DropDownItem{
+		{
+			Text: values.String(values.StrLimit),
+		},
+		{
+			Text: values.String(values.StrMarket),
+		},
+	}
+
+	vertical   = layout.Vertical
+	horizontal = layout.Horizontal
+)
+
+type DEXMarketPage struct {
+	*load.Load
+	// GenericPageModal defines methods such as ID() and OnAttachedToNavigator()
+	// that helps this Page satisfy the app.Page interface. It also defines
+	// helper methods for accessing the PageNavigator that displayed this page
+	// and the root WindowNavigator.
+	*app.GenericPageModal
+
+	scrollContainer *widget.List
+	ordersContainer *widget.List
+
+	serverSelector *cryptomaterial.DropDown
+	addServerBtn   *cryptomaterial.Clickable
+
+	marketSelector *cryptomaterial.DropDown
+
+	toggleBuyAndSellBtn *cryptomaterial.SegmentedControl
+	orderTypesDropdown  *cryptomaterial.DropDown
+	isMarketOrder       bool
```

Comment:
This should be a method rather than a property. `return pg.orderTypesDropdown.SelectedIndex() != limitOrderIndex`.

Or it can even be a local variable defined and assigned where it is needed.